pkg/controller/master/setting/nodeconfig.go

```diff
@@ -6,83 +6,151 @@ import (
-			if _, err := h.nodeConfigs.Create(newNodeConfig); err != nil {
-				logrus.Errorf("Create node config with nodeName %s failure: %v", nodeName, err)
-				return err
-			}
+			return err
+		}
+		lhsV2DataEngineCpy := lhsV2DataEngine.DeepCopy()
+		if enableV2DataEngine {
+			lhsV2DataEngineCpy.Value = "true"
 		} else {
-			logrus.Debugf("Nodeconfig content: %+v", nodeConfig)
-			nodeConfigCpy := nodeConfig.DeepCopy()
-			updateNodeConfig(nodeConfigCpy, parsedNTPServers)
-			if !reflect.DeepEqual(nodeConfig, nodeConfigCpy) {
-				if _, err := h.nodeConfigs.Update(nodeConfigCpy); err != nil {
-					logrus.Errorf("Update node config with nodeName %s failure: %v", nodeName, err)
-					return err
-				}
+			lhsV2DataEngineCpy.Value = "false"
+		}
+		if !reflect.DeepEqual(lhsV2DataEngine, lhsV2DataEngineCpy) {
+			// This will not work immediately if there's not enough hugepages,
+			// on each node, but because we return failure here when the
+			// setting doesn't apply, the change is continually requeued, so it
+			// will eventually go through once either the kubelets are restarted
+			// and pick up the new number of hugepages, or the nodes are rebooted
+			// and the allocation succeeds.
+			if _, err := h.longhornSettings.Update(lhsV2DataEngineCpy); err != nil {
+				return err
 			}
 		}
-
 	}
 
 	return nil
 }
 
-func updateNodeConfig(nodeConfig *nodev1.NodeConfig, ntpServers string) {
-	nodeConfig.Spec.NTPConfig.NTPServers = ntpServers
-}
+func (h *Handler) nodeOnChanged(_ string, node *corev1.Node) (*corev1.Node, error) {
+	if node == nil || node.DeletionTimestamp != nil {
+		return node, nil
+	}
 
-func generateNodeConfig(nodeName, ntpServers string) *nodev1.NodeConfig {
-	ntpConfig := generateNTPConfig(ntpServers)
-	return &nodev1.NodeConfig{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      nodeName,
-			Namespace: util.HarvesterSystemNamespaceName,
-		},
-		Spec: nodev1.NodeConfigSpec{
-			NTPConfig: ntpConfig,
-		},
+	// Get NTP Server settings
+	ntpServersSetting, err := h.settingCache.Get(harvSettings.NTPServersSettingName)
+	if err != nil {
+		return nil, err
+	}
+	ntpSettings := &util.NTPSettings{}
+	if err := json.Unmarshal([]byte(ntpServersSetting.Value), ntpSettings); err != nil {
+		return nil, fmt.Errorf("failed to parse NTP settings: %v", err)
+	}
+	ntpServers := util.ReGenerateNTPServers(ntpSettings)
+
+	// Figure out if we should enable the LH V2 data engine
+	enableV2DataEngineSetting, err := h.settingCache.Get(harvSettings.LonghornV2DataEngineSettingName)
+	if err != nil {
+		return nil, err
+	}
+	enableV2DataEngine := enableV2DataEngineSetting.Value == "true"
+	// Don't enable for witness nodes
+	if _, found := node.Labels[ctlnode.HarvesterWitnessNodeLabelKey]; found {
+		enableV2DataEngine = false
+	}
+	// Don't enable if explicitly disabled for this node
+	if value, found := node.Labels[longhorntypes.NodeDisableV2DataEngineLabelKey]; found && value == longhorntypes.NodeDisableV2DataEngineLabelKeyTrue {
```

Comment:
Just curious, is this label set up manually by the user?
Or do we have any mechanism to set up this label?